client/question/components/LearnerViewDirective.js

```diff
@@ -762,6 +757,32 @@ tie.directive('learnerView', [function() {
             tasks[currentTaskIndex].getId());
         };
 
+        /**
+         * Initializes the appropriate values in $scope for the question
+         * instructions, stored code, starter code and feedback.
+         *
+         * @param {string} questionId ID of question whose data will be loaded
+         */
+        $scope.loadQuestion = function(questionId) {
+          SessionIdService.resetSessionId();
+          if (SERVER_URL) {
+            try {
+              QuestionDataService.getQuestionAsync(questionId).then(
```

Comment:
Move server check into getQuestionAsync so that it's the only call here. (Look into q.defer)

Reply:
FYI, I read up a bit on this, and it looks like $q.resolve / $q.when might be a better replacement than $q.defer. See

  https://toddmotto.com/promises-angular-q#qwhen--qresolve
  http://www.codelord.net/2015/09/24/%24q-dot-defer-youre-doing-it-wrong/